src/main/java/algorithm/VisualizationManager.java

```diff
@@ -0,0 +1,395 @@
+    private double mouseOldX, mouseOldY = 0;
+    private final Rotate rotateX = new Rotate(0, Rotate.X_AXIS);
+    private final Rotate rotateY = new Rotate(0, Rotate.Y_AXIS);
+    private final Rotate rotateZ = new Rotate(0, Rotate.Z_AXIS);
+
+
+    public void injectStatusLabel(Label statusLabel) {
+        this.statusLabel = statusLabel;
+    }
+
+    public MeshView[] getMeshView() {
+        return stlFiles;
+    }
+
+    public ConeMesh[] getTrackingCones() {
+        return trackingCones;
+    }
+
+    /**
+     * Sets the reference to the ScrollPane.
+     *
+     * @param scrollPane new scrollPane
+     */
+    public void setPane(ScrollPane scrollPane) {
+        this.scrollPane = scrollPane;
+    }
+
+    /**
+     * Sets the reference to the MeshGroup.
+     *
+     * @param meshGroup new meshGroup
+     */
+    public void setMeshGroup(Group meshGroup) {
+        this.meshGroup = meshGroup;
+    }
+
+//    public void setViewportSize(double VIEWPORT_SIZE) {
+//        this.VIEWPORT_SIZE = VIEWPORT_SIZE;
+//        this.VIEWPORT_CENTER = VIEWPORT_SIZE / 2;
+//    }
+
+
+    /**
+     * Creates a Tracking Cone for each device
+     */
+    private void createTrackingCone() {
+        if (trackingService.getTrackingDataSource() == null) {
+            statusLabel.setText("No Tracking Data Source");
+            return;
+        }
+        statusLabel.setText("");
+
+        // update after CSV/IGT data has been loaded
+        trackingService.getTrackingDataSource().update();
+        ArrayList<Tool> tools = trackingService.getTrackingDataSource().getLastToolList();
+
+        if (tools != null) {
+            trackingCones = new ConeMesh[tools.size()];
+
+            for (int i = 0; i < trackingCones.length; i++) {
+                trackingCones[i] = new ConeMesh(12, 4, 8);
+            }
+        }
+    }
+
+    /**
+     * loads an STL File for Visualization
+     */
+    public void loadStlModel() {
+        StlMeshImporter importer = new StlMeshImporter();
+        FileChooser fc = new FileChooser();
+        fc.setTitle("Load STL File");
+        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("STL Files", "*.stl"));
+        List<File> fileList = fc.showOpenMultipleDialog(new Stage());
+
+        if (fileList != null) {
+            stlFiles = new MeshView[fileList.size()];
+            int x = 0;
+
+            for (File file : fileList) {
+                try {
+                    importer.read(file);
+                    Mesh mesh = importer.getImport();
+                    stlFiles[x] = new MeshView(mesh);
+
+                    x++;
+
+                } catch (Exception e) {
+                    logger.log(Level.WARNING, "Error reading STL file");
+                }
+            }
+        }
+    }
+
+    /**
+     * Start the visualization of tracking data that is read from CSV or IGT connection
+     */
+    public void startVisualization() {
```

Comment:
Since this method is not really used to *start* the visualization, but rather to update it constantly, I'd suggest to rename it accordingly to avoid confusion

Reply:
renamed visualizeTracking